databricks/koala/structures.py

```diff
@@ -239,6 +239,18 @@ def _metadata(self):
             self._pandas_metadata = ref._metadata.copy(column_fields=[self.name])
         return self._pandas_metadata
 
+    @_metadata.setter
+    def _metadata(self, metadata):
```

Comment:
instead of setters, get we just have `_set_metadata`? These functions are not meant to be user-facing, there is no advantage in having a setter.

Reply:
Sure, I don't have a preference for that. Should we replace with it for `DataFrame` as well?